src/picai_prep/dcm2mha.py

```diff
@@ -520,9 +524,45 @@ def _read_image(self, path: Optional[PathLike] = None) -> sitk.Image:
         try:
             return self._read_image_sitk(path=path)
         except RuntimeError:
-            return self._read_image_pydicom(path=path)
+            # try again while removing localizer slices
+            try:
+                return self._read_image_sitk(path=path, filter_localizer_slices=True)
+            except RuntimeError:
+                # try again with pydicom
+                return self._read_image_pydicom(path=path)
+
+    @staticmethod
+    def _filter_localizer_slices(dicom_slice_paths: List[str]) -> List[str]:
```

Comment:
I understand the reasoning, but I find it very ugly. Is there a way to neatly combine this with lines 615-619?

Reply:
Fair enough. Thinking about it a bit more resulted in this way: https://github.com/DIAGNijmegen/picai_prep/pull/40
What do you think?